### PR TITLE
Add protocol values to getversion

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo-modules/issues/636

### CODE REVIEW

**Comment by cloud8little** [comment]
seems Address Version should also be included, @shargon

besides, there are other parameters, maybe they are also needed. @devhawk what do you think?

1. MaxTransactionsPerBlock
2. MemoryPoolMaxTransactions
3. InitialGasDistribution

**Comment by shargon** [comment]
If we decide which parameters should be added, we can do it now :)

**Comment by devhawk** [comment]
> besides, there are other parameters, maybe they are also needed. @devhawk what do you think?

It would seem reasonable that `getversion` returns all of these values. I'm not sure how often they would be used, but I don't think there's much in the way of additional incremental cost to include them

**Comment by cloud8little** [comment]
> > besides, there are other parameters, maybe they are also needed. @devhawk what do you think?
> 
> It would seem reasonable that `getversion` returns all of these values. I'm not sure how often they would be used, but I don't think there's much in the way of additional incremental cost to include them

Alright, let's add them @shargon

File: src/RpcClient/Models/RpcVersion.cs [hunk 1]
```diff
@@ -14,27 +44,28 @@ public class RpcVersion
 
         public string UserAgent { get; set; }
 
+        public RpcProtocol Protocol { get; set; } = new();
 
         public JObject ToJson()
         {
             JObject json = new();
-            json["network"] = Network;
```
Comment (roman-khimov) [inline src/RpcClient/Models/RpcVersion.cs:21]: This will break compatibility, do we really want that?

Comment (shargon) [reply src/RpcClient/Models/RpcVersion.cs:21]: We can create `getprotocol`

Comment (roman-khimov) [reply src/RpcClient/Models/RpcVersion.cs:21]: I'd rather just return `network` here and in `protocol` subsection, make a notice that top-level `network` is deprecated and will be removed and then remove it in some subsequent release.

File: src/RpcServer/RpcServer.Node.cs [hunk 2]
```diff
@@ -62,7 +62,15 @@ protected virtual JObject GetVersion(JArray _params)
             json["wsport"] = localNode.ListenerWsPort;
             json["nonce"] = LocalNode.Nonce;
             json["useragent"] = LocalNode.UserAgent;
-            json["network"] = system.Settings.Network;
+
```
Comment (cloud8little) [inline src/RpcServer/RpcServer.Node.cs:65]:
```suggestion
json["protocol"] = new JObject();
```

**Review by cloud8little — APPROVED** [review]

**Comment by cloud8little** [comment]
Test passed. wait for @erikzhang review.

**Review by erikzhang — APPROVED** [review]

**Pull request merged**